projects/pswg/src/main/java/com/parzivail/pswg/entity/ThrowableExplosive.java

```diff
@@ -177,27 +176,23 @@ public boolean damage(DamageSource source, float amount)
 				this.delay = 2;
 				this.shouldExplode = true;
 			}
-
-			return true;
 		}
 		else if (source.isIn(SwgTags.DamageTypes.IGNITES_EXPLOSIVE))
 			if (!this.shouldExplode)
 				this.explode();
 
 		return super.damage(source, amount);
 	}
-
 	public void explode()
 	{
-		if (!this.getWorld().isClient)
+		if (!this.getWorld().isClient())
```

Comment:
is this a method? if so, why is this being used like a method here and used like a field elsewhere?

Reply:
Also, if you're going to be casting later to `ServerWorld`, you might want to use `if (this.getWorld instanceof ServerWorld serverWorld)` and use that newly-defined variable `serverWorld` in the if's body. For more info: [Pattern Matching in Java](https://docs.oracle.com/en/java/javase/17/language/pattern-matching-instanceof-operator.html#GUID-843060B5-240C-4F47-A7B0-95C42E5B08A7)

Reply:
Not sure about the .isClient thing. Just one of the quirks of minecraft I think

Reply:
Did it not work using the field? Did using the method make it behave differently?